streams/src/test/java/org/apache/kafka/streams/kstream/internals/KStreamKStreamLeftJoinTest.java

```diff
@@ -88,7 +90,22 @@ public void testLeftJoinWithInvalidSpuriousResultFixFlag() {
     }
 
     @Test
-    public void testLeftJoinWithSpuriousResultFixDisabled() {
+    public void testLeftJoinWithSpuriousResultFixDisabledViaFeatureFlag() {
+        runLeftJoinWithoutSpuriousResultFix(
+            JoinWindows.ofTimeDifferenceAndGrace(ofMillis(100L), ofHours(24L)),
+            false
+        );
+    }
+    @Test
+    public void testLeftJoinWithSpuriousResultFixDisabledOldApi() {
```

Comment:
I duplicate this test, to verify that the feature flag, as well as the old API disables this fix. Thus, the usage of the old API in this method should not be changes via KIP-633 PR.